packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart

```diff
@@ -119,7 +119,10 @@ abstract class VideoPlayerPlatform {
   void _verifyProvidesDefaultImplementations() {}
 }
 
+/// Description of the data source used to create an instance of
+/// the video player.
 class DataSource {
+  /// Constructs an instance of [DataSource].
```

Comment:
Maybe this could mention which of the params are okay to be null and what some valid values look like for the string ones? Same for the other constructor.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#writing-prompts-for-good-documentation

Reply:
Added some more more info to the constructor documentation.